ctapipe/calib/camera/r1.py

```diff
@@ -191,6 +192,144 @@ def calibrate(self, event):
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool : ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        kwargs
+        """
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        self.telid = 0
+        self.pedestal_value_array = None
+        self.n_pixels = 7
+        self.size4drs = 4 * 1024
+        self.roisize = 40
+        self.offset = 300
+        self.high_gain = 0
+        self.low_gain = 1
+
+        self._load_calib()
+
+    def calibrate(self, event):
+        """
+        Perform calibration on event using pedestal file.
+        If pedestal file is wrong raise ValueError Exception.
+        Parameters
+        ----------
+        event : `ctapipe` event-container
+        """
+        samples = np.zeros((event.r0.tel[self.telid].waveform.shape), dtype=np.uint16)
+        event.r1.tel[self.telid].waveform = samples
+
+        event_number_of_modules = event.lst.tel[0].svc.num_modules
+        if event_number_of_modules == self.number_of_modules_from_file:
+            for nr_module in range(0, event_number_of_modules):
+                first_cap = self._get_first_capacitor(event, nr_module)
+                for gain in range(0, 2):
+                    for pixel in range(0, self.n_pixels):
+                        for roi in range(0, self.roisize):
+                            position = int((roi + first_cap[gain, pixel]) % self.size4drs)
+                            val = (
+                                event.r0.tel[0].waveform[gain, pixel + nr_module * 7, roi]
+                                - int(self.pedestal_value_array[nr_module, gain,
+                                                                pixel, position])
+                                  ) + self.offset
+                            event.r1.tel[self.telid].waveform[
+                                gain, pixel + nr_module * 7, roi] = val
+        else:
+            self.log.error("No match number of modules {} in pedestal file,"
+                             " to number of modules {} in event-container. "
+                             "Check if path to pedestal file is correct. ".
+                             format(self.number_of_modules_from_file,
+                                    event_number_of_modules))
+
+            raise ValueError("Wrong pedestal file")
+
+
+    def _load_calib(self):
+        """
+        If a pedestal file has been supplied, create a array with
+        pedestal value . If it hasn't then point calibrate to
+        fake_calibrate, where nothing is done to the waveform.
+        """
+        Provenance().add_input_file(self.pedestal_path)
+        if self.pedestal_path:
+            with open(self.pedestal_path, "rb") as binary_file:
```

Comment:
can you use a standard file format, preferably a FITS table for this rather than a custom binary format? We would like to avoid any hand-coded files, and soon will also require a set of metadata headers that must be included in any file. That's difficult with a custom format.

Reply:
We would like use the same file format as camera server can use, and unfortunately camera server can't use FITS table (because zFits decoder is not implement inside the camera server). The best for this reason would be simply binary file. What data format is allowed in ctapipe?

Reply:
> FITS table (because zFits decoder is not implement inside the camera server)

Why not use a standard-conform fits file that is readable by e.g. libcfitsio (and thus also astropy)?

Using custom binary formats never turned out well in the end for maintainability.

Reply:
Thanks a lot for comments. You are right. I will use fits file to store pedestal values.

Reply:
I'd suggest just a ECSV text file then - that can be read by essentially everything, including astropy.table.  It's what we already use for many files in ctapipe. 

http://docs.astropy.org/en/stable/api/astropy.io.ascii.Ecsv.html